### ISSUE TITLE
Custom hostname doesn't work when using https based origin

### ISSUE THREAD

**Issue by KORDayDream** [git-town/git-town#2188]
Hello,

I am currently attempting to use Git Town with our enterprise GitHub account.

After some investigation, I have found that setting the driver and origin hostname does not appear to enable Git Town to connect to our remote when using an HTTPS-based origin. Our company's GitHub hostname is formatted as "repo.companyname.com" and does not include "github" in the hostname.

When I connected to the remote via SSH, Git Town worked perfectly fine. Therefore, I assume that the custom hostname only works for SSH connections. This belief was reinforced after reviewing the "features/git-town-new-pull-request/self_hosted.feature" file.

Am I missing something? Or is this currently the normal behavior?

In addition, I have also observed that Git Town's "ship" command attempts to call "api.github.com" instead of our enterprise GitHub hostname. According to the answer provided in this StackOverflow post (https://stackoverflow.com/a/50612869), I need to call "repo.companyname.com/api/v3/..." instead. While this appears to be a separate issue from my previous inquiry(and is also a duplicate: #1179), I hope that it can also be addressed as it would greatly benefit users operating within a GitHub enterprise environment.

Thank you for your help.

**Comment by kevgo** [comment]
Hello @KORDayDream, have you set the [code-hosting-driver](https://www.git-town.com/preferences/code-hosting-driver.html) setting? To do so, please run this command on your machine:

```
git config git-town.code-hosting-driver github
```

This should make the `git new-pull-request` and `git repo` commands work with your setup. The respective end-to-end test that describes/verifies how this works is https://github.com/git-town/git-town/blob/main/features/new_pull_request/features/self_hosted.feature.

`git ship` might still be broken. Please upvote https://github.com/git-town/git-town/issues/1179 to help the Git Town team prioritize fixing this bug. Thanks! :bow:

**Comment by KORDayDream** [comment]
Hello @kevgo ! Thank you for your response. 

I appreciate your suggestion, but unfortunately I have already tried the command `git config --global git-town.code-hosting-driver github` along with `git config --global git-town.code-hosting-origin-hostname repo.companyname.com` but it did not resolve the issue I'm facing.

I believe that there might be other factors causing this problem, and I would appreciate it if you could provide me some further suggestions on how to solve this issue.

Thank you for your help!

**Comment by kevgo** [comment]
Sorry for the problems you encounter. I'm happy to help (and fix a few Git Town bugs along the way). To investigate your setup:

1. Please run `git config -l` and post the result here. Please obfuscate sensitive information like access tokens.

2. Post specific Git Town commands that fail, and their output.

Thanks!

**Comment by KORDayDream** [comment]
I am sorry to tell you that upon further investigation, I discovered that the problem was caused by internal issues such as the internal github server being down, DNS problems, and other related issues.

I have tested all the features, except git town ship, that connect to the remote origin, and found that they are working properly.

Thank you for your valuable time and assistance. I will be closing this issue now.
